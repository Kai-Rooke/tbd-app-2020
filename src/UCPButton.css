

  .smallbutton{
    background-color: #31B7BC;
    color: black;
    width: 100px;
    height: 50px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }

  

  .mediumbutton{
    background-color: #31B7BC;
    color: black;
    width: 125px;
    height: 75px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }


