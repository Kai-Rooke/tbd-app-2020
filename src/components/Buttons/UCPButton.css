
  .smallbutton{
    background-color: #31B7BC;
    color: black;
    width: 100px;
    height: 50px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }
  .mediumbutton{
    background-color: #31B7BC;
    color: black;
    width: 125px;
    height: 75px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }
  .largebutton{
    background-color: #31B7BC;
    color: black;
    width: 200px;
    height: 100px; 
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }

  .extralargebutton{
    background-color: #31B7BC;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
  }
.finalbutton{
  background-color: #31B7BC;
  color: black;
  width: 100px;
  height: 50px;
  font-size: large;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  border-radius: 25px;
  border: 2px solid #005C6E;
}
