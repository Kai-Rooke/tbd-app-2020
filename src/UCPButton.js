import React from 'react';
import PropTypes from 'prop-types';
import './UCPButton.css';
export const UCPButton =
({primary, className, buttonText, ...props}) => {
    return (<button type="button" className={className}{...props}>{buttonText}</button>);

};

UCPButton.propTypes = {
primary:PropTypes.bool,
className:PropTypes.string,
onClick:PropTypes.func,
};

UCPButton.defaultProps = {
    primary:false, size: 'UCPDefault', onClick:undefined
};

export default UCPButton ;