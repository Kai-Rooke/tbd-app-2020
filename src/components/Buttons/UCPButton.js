import React from 'react';
import PropTypes from 'prop-types';
import './UCPButton.css';

export const UCPButton = ({ className, buttonText, ...props}) => {
     return (
    <button data-testid="ucpButton"
     type="button"
     className={className}
          {...props}
     >{buttonText}â€‹
    </button>
  );
};
UCPButton.propTypes = {
    /**
   * Is this the principal call to action on the page?
   */
className:PropTypes.string,
/**
   * Optional click handler
   */
onClick:PropTypes.func,
};
UCPButton.defaultProps = {
    size:'UCPDefault',
    onClick:undefined
};
export default UCPButton

/* Example

<UCPButton primary=true className="Large" buttonText="Click Here"/> */
