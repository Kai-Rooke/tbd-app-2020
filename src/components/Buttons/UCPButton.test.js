import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import UCPButton from './UCPButton';

//cleans up 
afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<UCPButton className="largetealbutton" buttonText="Click Here"/>);
    expect(asFragment()).toMatchSnapshot();
});

it("Button Test Id", () => {
    const { getByTestId } = render(<UCPButton className="largetealbutton" buttonText="Click Here"/>);
    expect(getByTestId('ucpButton'));
});

/*
// NOTE: jest-dom adds handy assertions to Jest and it is recommended, but not required.
import '@testing-library/jest-dom/extend-expect'

import { render, fireEvent } from '@testing-library/svelte'

import Comp from '../Comp'

test('shows proper heading when rendered', () => {
  const { getByText } = render(Comp, { name: 'World' })

  expect(getByText('Hello World!')).toBeInTheDocument()
})

// Note: This is as an async test as we are using `fireEvent`
test('changes button text on click', async () => {
  const { getByText } = render(Comp, { name: 'World' })
  const button = getByText('Button')

  // Using await when firing events is unique to the svelte testing library because
  // we have to wait for the next `tick` so that Svelte flushes all pending state changes.
  await fireEvent.click(button)

  expect(button).toHaveTextContent('Button Clicked')
})
*/