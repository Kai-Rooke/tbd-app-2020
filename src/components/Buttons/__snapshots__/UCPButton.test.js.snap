// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <a
    href="/emptyLink"
  >
    <button
      class="largetealbutton"
      data-testid="ucpButton"
      type="button"
    >
      Click Hereâ€‹
    </button>
  </a>
</DocumentFragment>
`;
